src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm_copy_b.cpp

```diff
@@ -0,0 +1,315 @@
+    return ss.str();
+}
+std::string BrgemmCopyBKernelConfig::StaticParams::to_string() const {
+    std::stringstream ss;
+    PRINT(src_dt); PRINT(wei_dt); PRINT(isa);
+    PRINT(is_with_comp); PRINT(is_transposed_B); PRINT(wei_N_blk);
+    return ss.str();
+}
+#undef PRINT
+#endif
+
+BrgemmCopyBKernel::BrgemmCopyBKernel() : jit_generator(jit_name()), ker_(nullptr) {}
+
+BrgemmCopyBKernel::BrgemmCopyBKernel(const BrgemmCopyBKernelConfig& conf)
+    : jit_generator(jit_name()), ker_(nullptr), conf(conf), kernel(create_brgemm_copy_b_kernel()) {
+    OV_CPU_JIT_EMITTER_ASSERT(kernel, "Kernel is missed!");
+}
+
+status_t BrgemmCopyBKernel::create_kernel() {
+    const auto code = jit_generator::create_kernel();
+    OV_CPU_JIT_EMITTER_ASSERT(code == status::success, "Failed to create kernel");
+    ker_ = (decltype(ker_))jit_ker();
+    return code;
+}
+
+void BrgemmCopyBKernel::operator()(const call_args* args) const {
+    OV_CPU_JIT_EMITTER_ASSERT(ker_, "Kernel is nullptr");
+    ker_(args);
+}
+
+void BrgemmCopyBKernel::generate() {
+    preamble();
+
+    mov(src_reg, ptr[abi_param1 + GET_OFF_BRGEMM_COPY_B_ARGS(src)]);
+    mov(tr_src_reg, ptr[abi_param1 + GET_OFF_BRGEMM_COPY_B_ARGS(tr_src)]);
+    if (conf.is_with_comp())
+        mov(comp_reg, ptr[abi_param1 + GET_OFF_BRGEMM_COPY_B_ARGS(compensation_ptr)]);
+
+    const size_t data_size = dnnl_data_type_size(conf.get_wei_dt());
+    const size_t brgemmVNNIFactor = data_type_vnni_granularity(conf.get_wei_dt());
+    size_t start_in = 0;
+    size_t start_out = 0;
+    size_t start_comp = 0;
+
+    auto add_ptr_increments = [&](size_t current_N) {
+        start_in += conf.is_transposed_B() ? conf.get_K() * current_N * data_size : current_N * data_size;
+        start_out += current_N * brgemmVNNIFactor * data_size;
+        start_comp += conf.is_with_comp() ? current_N * sizeof(int32_t) : 0;
+    };
+
+    // OneDNN requires tail handling before main iterations
+    if (conf.get_wei_N_tail() != 0) {
+        emit_brgemm_copy_b_kernel_call(conf.get_wei_N_tail(), conf.get_K(), start_in, start_out, start_comp);
+        add_ptr_increments(conf.get_wei_N_tail());
+    }
+
+    for (auto nb = conf.get_wei_N_tail(); nb < conf.get_N_blk(); nb += conf.get_wei_N_blk()) {
+        emit_brgemm_copy_b_kernel_call(conf.get_wei_N_blk(), conf.get_K(), start_in, start_out, start_comp);
+        add_ptr_increments(conf.get_wei_N_blk());
+    }
+
+    postamble();
+}
+
+std::shared_ptr<BrgemmCopyBKernel::dnnl_brgemm_copy_b_kernel> BrgemmCopyBKernel::create_brgemm_copy_b_kernel() const {
+    matmul::brgemm_matmul_conf_t brgCopyKernelConf;
+    brgCopyKernelConf.src_dt = conf.get_src_dt();
+    brgCopyKernelConf.wei_dt = conf.get_wei_dt();
+    brgCopyKernelConf.orig_wei_dt = brgCopyKernelConf.wei_dt;
+    brgCopyKernelConf.wei_n_blk = static_cast<int>(conf.get_wei_N_blk());
+    // However, the generated primitive can be also applied to tensors with other ranks
```

Comment:
May I ask you to transfer the whole comment from copy_b emitter?
```suggestion
    // Note: 2D format tags are used just to force the needed OneDNN primitive creation.
    // However, the generated primitive can be also applied to tensors with other ranks
```

Reply:
Sure! Looks like I accidentally removed first line. Done, thanks!